### PR TITLE
Reqeust new kubeconfig when user's groups changed

### LINKED ISSUE #2940: Feat: support user group
Members of a user group inherit that group's permissions and also retain any permissions they have separately.

/kind feature-request
/area iam

### PR DESCRIPTION
**Pull request by RolandMa1986**
Signed-off-by: Roland.Ma <rolandma@yunify.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
When a user assigned to a group. The user's kubeconfig should be updated too.

Fixes #2940

/assign @wansir

### CODE REVIEW

File: pkg/models/kubeconfig/kubeconfig.go [hunk 1]
```diff
@@ -140,9 +148,19 @@ func (o *operator) CreateKubeConfig(user *iamv1alpha2.User) error {
...
+	cm = &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       configMapKind,
+			APIVersion: configMapAPIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: configName,
+			Labels: map[string]string{
+				constants.UsernameLabelKey:      user.Name,
+				iamv1alpha2.GroupReferenceLabel: groups,
```
Comment (wansir) [inline pkg/models/kubeconfig/kubeconfig.go:160]: Label must be no more than 63 characters.

File: pkg/models/kubeconfig/kubeconfig.go [hunk 2]
```diff
@@ -79,17 +81,23 @@ func (o *operator) CreateKubeConfig(user *iamv1alpha2.User) error {
...
-	if !errors.IsNotFound(err) {
-		klog.Error(err)
-		return err
+	if err == nil {
+		if cm.Labels[iamv1alpha2.GroupReferenceLabel] == groups {
+			// already exist and groups are same.
+			return nil
+		}
+		// remove the old configmap if groups are changed.
+		o.k8sClient.CoreV1().ConfigMaps(constants.KubeSphereControlNamespace).Delete(configName, &metav1.DeleteOptions{})
```
Comment (wansir) [inline pkg/models/kubeconfig/kubeconfig.go:94]: the error must be handled

**Comment by zryfish** [comment]
What's the status of this PR, are we going to working on this later? If not so, we'd better close.

**Comment by RolandMa1986** [comment]
/close
Kubernetes doesn't provide a way to revoke cert. If we singed cert with groups' permission, there's a potential security risk. 
See: https://github.com/kubernetes/kubernetes/issues/18982

**Pull request closed without merging**